<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Read a GenBank File — readGenBank • rOpenSci: genbankr</title>

<!-- favicons -->
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="/apple-touch-icon.png">

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>


<!-- docsearch -->
<script src="../docsearch.js"></script>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous" />
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>


<!-- Bootstrap -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.1/css/all.min.css" integrity="sha256-PbSmjxuVAzJ6FPvNYsrXygfGhNJYyZ2GktDbkMBqQZg=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.1/css/v4-shims.min.css" integrity="sha256-A6jcAdwFD48VMjlI3GDxUd+eCQa7/KWy6G9oe/ovaPA=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>

<!-- ropensci -->
<link href="../ropensci.css" rel="stylesheet">

<meta property="og:title" content="Read a GenBank File — readGenBank" />
<meta property="og:description" content="Read a GenBank file from a local file, or retrieve
and read one based on an accession number. See Details for exact
behavior." />
<meta name="twitter:card" content="summary" />






<!-- Matomo -->
<link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.css" />
<script src="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.js" data-cfasync="false"></script>
<script src="https://ropensci.org/scripts/matomo.js"></script>
<noscript><p><img src="https://ropensci.matomo.cloud/matomo.php?idsite=1&amp;rec=1" style="border:0;" alt="" /></p></noscript>

<!-- End Matomo Code -->

  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a href="https://ropensci.org"><img src="https://ropensci.org/img/icon_short_white.svg" id="hexlogo" alt="rOpenSci" /></a>
        <a class="navbar-brand" href="../index.html"><i>genbankr <small>v1.21.0</small></i></a>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/genbankr.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
        
      </ul>
      
      <form class="navbar-form navbar-right docsearch" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
        </div>
      </form>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Read a GenBank File</h1>
    
    <div class="hidden name"><code>readGenBank.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Read a GenBank file from a local file, or retrieve
and read one based on an accession number. See Details for exact
behavior.</p>
    </div>

    <pre class="usage"><span class='fu'>readGenBank</span><span class='op'>(</span><span class='va'>file</span>, text <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/readLines.html'>readLines</a></span><span class='op'>(</span><span class='va'>file</span><span class='op'>)</span>, partial <span class='op'>=</span> <span class='cn'>NA</span>, ret.seq <span class='op'>=</span> <span class='cn'>TRUE</span>,
  verbose <span class='op'>=</span> <span class='cn'>FALSE</span><span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>file</th>
      <td><p>character or GBAccession. The path to the file, or a GBAccession object containing Nuccore versioned accession numbers. Ignored if <code>text</code> is specified.</p></td>
    </tr>
    <tr>
      <th>text</th>
      <td><p>character. The text of the file. Defaults to text within <code>file</code></p></td>
    </tr>
    <tr>
      <th>partial</th>
      <td><p>logical. If TRUE, features with non-exact boundaries will
be included. Otherwise, non-exact features are excluded, with a warning
if <code>partial</code> is <code>NA</code> (the default).</p></td>
    </tr>
    <tr>
      <th>ret.seq</th>
      <td><p>logical. Should an object containing the raw ORIGIN sequence be
created and returned. Defaults to <code>TRUE</code>. If <code>FALSE</code>,
the sequence slot is set to <code>NULL</code>. See NOTE.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>logical. Should informative messages be printed to the
console as the file is processed. Defaults to <code>FALSE</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A <code>GenBankRecord</code> object containing (most,
see detaisl) of the information within <code>file</code>/<code>text</code> Or a list of
<code>GenBankRecord</code> objects in cases where a
<code>GBAccession</code> vector with more than one ID in it is passed to <code>file</code></p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>If a a <code>GBAccession</code> object is passed to <code>file</code>, the
rentrez package is used to attempt to fetch full GenBank records for all
ids in the</p>
<p>Often times, GenBank files don't contain exhaustive annotations.
For example, files including CDS annotations often do not have separate
transcript features.  Furthermore, chromosomes are not always named,
particularly in organisms that have only one. The details of how genbankr
handles such cases are as follows:</p>
<p>In files where CDSs are annotated but individual exons are not, 'approximate
exons' are defined as the individual contiguous elements within each CDS.
Currently, no mixing of approximate and explicitly annotated exons is
performed, even in cases where, e.g., exons are not annotated for some
genes with CDS annotations.</p>
<p>In files where transcripts are not present, 'approximate transcripts'
defined by the ranges spanned by groups of exons are used.  Currently, we do
not support generating approximate transcripts from CDSs in files that
contain actual transcript annotations, even if those annotations do not
cover all genes with CDS/exon annotations.</p>

<p>Features  (gene, cds, variant, etc) are assumed to be contained within the
most recent previous source feature (chromosome/physical piece of DNA).
Chromosome name for source features (seqnames in the resulting
<code>GRanges</code>/<code>VRanges</code> is determined  as follows:</p><ol>
<li><p>The 'chromosome' attribute, as is (e.g., "chr1");</p></li>
<li><p>the 'strain' attribute, combined with auto-generated count (e.g.,
"VR1814:1");</p></li>
<li><p>the 'organism' attribute, combined with auto-generated count (e.g.
"Human herpesvirus 5:1".</p></li>
</ol>

<p>In files where no origin sequence is present, importing varation
features is not currently supported, as there is no easy/
self-contained way of determining the reference in those
situations and the features themselves list only alt. If
variation features are present in a file without origin
sequence, those features are ignored with a warning.</p>
<p>Currently some information about from the header of a GenBank file,
primarily reference and author based information, is not captured and
returned. Please contact the maintainer if you have a direct use-case for
this type of information.</p>
    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>We have endeavored to make this parser as effcient as easily possible.
On our local machines, a 19MB genbank file takes 2-3 minutes to be parsed.
That said, this function is not tested and likely not suitable for
processing extremely large genbank files.</p>
<p>The origin sequence is always parsed when calling readGenBank, because
it is necessary to generate a VRanges from variant features. So currently
<code>ret.seq=FALSE</code> will not reduce parsing time, or maximum memory usage,
though it will reduce memory usage by the final GenBankRecord object. The
lower-level parseGenBank does skil parsing the sequence at all via
<code>ret.seq=FALSE</code>, but variant annotations will be excluded if
make_gbrecord is called on the resulting parsed list.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='va'>gb</span> <span class='op'>=</span> <span class='fu'>readGenBank</span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/base/system.file.html'>system.file</a></span><span class='op'>(</span><span class='st'>"sample.gbk"</span>, package<span class='op'>=</span><span class='st'>"genbankr"</span><span class='op'>)</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Annotations don't have 'locus_tag' label, using 'gene' as gene_id column</span></div><div class='output co'>#&gt; <span class='message'>Annotations don't have 'locus_tag' label, using 'gene' as gene_id column</span></div><div class='output co'>#&gt; <span class='message'>No exons read from genbank file. Assuming sections of CDS are full exons</span></div><div class='output co'>#&gt; <span class='message'>No transcript features (mRNA) found, using spans of CDSs</span></div><div class='output co'>#&gt; <span class='warning'>Warning: Got unexpected multi-value field(s) [ rpt_type ]. The resulting column(s) will be of class CharacterList, rather than vector(s). Please contact the maintainer if multi-valuedness is expected/meaningful for the listed field(s).</span></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <!-- begin footer -->
<div class="footer">
    <div class="container">
        <div class="row start bottom-8">
            <div class="col-xs-2"> <img id="footerlogo" src="https://ropensci.org/img/icon_short_white.svg" /></div>
            <div class="col-xs-10">
                <div class="row">
                    <div class="col-md-4 col-xs-6">
                        <a href="http://github.com/ropensci" target="_blank"><div class="icon fab fa-github"></div></a>
                        <a href="http://github.com/ropenscilabs" target="_blank"><div class="icon fa fa-flask"></div></a>
                        <a href="http://twitter.com/ropensci" target="_blank"><div class="icon fab fa-twitter"></div></a>
                        <a href="http://vimeo.com/ropensci" target="_blank"><div class="icon fab fa-vimeo"></div></a>
                    </div>
                </div>
                <div class="row top-4">
                    <div class="col-md-2 col-sm-4">
                        <ul>
                            <h5 class="bottom-2">About</h5>
                            <li><a href="https://ropensci.org/about">About rOpenSci</a></li>
                            <li><a href="https://ropensci.org/software-review">Software Review Process</a></li>
                            <li><a href="https://ropensci.org/about#team">Our Team</a></li>
                            <li><a href="https://ropensci.org/careers">Jobs</a></li>
                            <li><a href="https://ropensci.org/donate">Donate</a></li>
                            <li><a href="https://ropensci.org/contact">Contact Us</a></li>
                        </ul>
                    </div>
                    <div class="col-md-3 col-sm-4">
                        <ul>
                            <h5 class="bottom-2">Community</h5>
                            <li><a href="https://ropensci.org/community/">Our Community</a></li>
                            <li><a href="https://ropensci.org/commcalls/">Community calls</a></li>
                            <li><a href="https://ropensci.org/events/">Events</a></li>
                            <li><a href="https://discuss.ropensci.org/">Join the Discussion</a></li>
                            <li><a href="https://ropensci.org/code-of-conduct">Code of conduct</a></li>
                        </ul>
                    </div>
                    <div class="col-md-2 col-sm-4">
                        <ul>
                            <h5 class="bottom-2">Resources</h5>
                            <li><a href="https://ropensci.org/packages/">Packages</a></li>
                            <li><a href="https://ropensci.org/usecases/">Use Cases</a></li>
                            <li><a href="https://ropensci.org/talks-papers/">Talks & Publications</a></li>
                            <li><a href="https://docs.ropensci.org/">Documentation</a></li>
                            <li><a href="https://ropensci.org/how-to-cite-ropensci/">Cite rOpenSci</a></li>
                        </ul>
                    </div>
                    <div class="col-md-4 col-xs-12">
                        <h5 class="bottom-2"></h5>

                        <p>rOpenSci is a fiscally sponsored project of <a href="http://numfocus.org" target="_blank">NumFOCUS</a></p>
                        <a href="http://numfocus.org" target="_blank"></a>

                    </div>
                </div>
            </div>
        </div>
    </div>
</div>
<!-- / end footer -->

      </footer>
   </div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script>
<script>
  docsearch({
    
    
    apiKey: '799829e946e1f0f9cd5b5a782c6316b9',
    indexName: 'ropensci-genbankr',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>



  </body>
</html>


