<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title> • rOpenSci: genbankr</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="/apple-touch-icon.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- docsearch --><script src="../docsearch.js"></script><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.3/docsearch.min.css" integrity="sha256-QOSRU/ra9ActyXkIBbiIB144aDBdtvXBcNc3OTNuX/Q=" crossorigin="anonymous">
<link href="../docsearch.css" rel="stylesheet">
<script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.1/css/all.min.css" integrity="sha256-PbSmjxuVAzJ6FPvNYsrXygfGhNJYyZ2GktDbkMBqQZg=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.1/css/v4-shims.min.css" integrity="sha256-A6jcAdwFD48VMjlI3GDxUd+eCQa7/KWy6G9oe/ovaPA=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><!-- ropensci --><link href="../ropensci.css" rel="stylesheet">
<meta property="og:title" content="">
<meta property="og:description" content="genbankr">
<meta name="twitter:card" content="summary">
<!-- Matomo --><link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.css">
<script src="https://cdn.jsdelivr.net/npm/cookieconsent@3/build/cookieconsent.min.js" data-cfasync="false"></script><script src="https://ropensci.org/scripts/matomo.js"></script><noscript><p><img src="https://ropensci.matomo.cloud/matomo.php?idsite=1&amp;rec=1" style="border:0;" alt=""></p></noscript>
<!-- End Matomo Code -->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>

      <div class="navbar-brand-container">
        <a href="https://ropensci.org"><img src="https://ropensci.org/img/icon_short_white.svg" id="hexlogo" alt="rOpenSci"></a>
        <a class="navbar-brand" href="../index.html"><i>genbankr <small>v1.21.0</small></i></a>
      </div>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav navbar-right">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/genbankr.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
        
      </ul>
<form class="navbar-form navbar-right docsearch" role="search">
        <div class="form-group">
          <input type="search" class="form-control" name="search-input" id="search-input" placeholder="Search..." aria-label="Search for..." autocomplete="off">
</div>
      </form>
      
    </div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip></h1>
            
      
      
      <div class="hidden name"><code>genbankr.Rmd</code></div>

    </div>

    
    
<div id="basic-usage" class="section level1">
<h1 class="hasAnchor">
<a href="#basic-usage" class="anchor"></a>Basic usage</h1>
<p>Simply put, the <code>genbankr</code> package parses files in the NCBI’s GenBank (gb/gbk) format into R.</p>
<p>The primary workhorse is provided by the <code>readGenBank</code> function, which accepts either a GenBank file (via the <code>file</code> argument) or raw text in the GenBank format (via the <code>text</code> argument).</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">genbankr</span><span class="op">)</span><span class="op">)</span>
<span class="va">smpfile</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"sample.gbk"</span>, package<span class="op">=</span><span class="st">"genbankr"</span><span class="op">)</span>
<span class="va">gb</span> <span class="op">=</span> <span class="fu"><a href="../reference/readGenBank.html">readGenBank</a></span><span class="op">(</span><span class="va">smpfile</span><span class="op">)</span></code></pre></div>
<pre><code>## Annotations don't have 'locus_tag' label, using 'gene' as gene_id column
## Annotations don't have 'locus_tag' label, using 'gene' as gene_id column</code></pre>
<pre><code>## No exons read from genbank file. Assuming sections of CDS are full exons</code></pre>
<pre><code>## No transcript features (mRNA) found, using spans of CDSs</code></pre>
<pre><code>## Warning in fill_stack_df(feats[!typs %in% c("gene", "exon", "CDS",
## "variation", : Got unexpected multi-value field(s) [ rpt_type ]. The resulting
## column(s) will be of class CharacterList, rather than vector(s). Please contact
## the maintainer if multi-valuedness is expected/meaningful for the listed
## field(s).</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gb</span></code></pre></div>
<pre><code>## GenBank Annotations
## Human herpesvirus 5 strain VR1814, complete genome. 
## Accession: GU179289
## 1 Sequence(s) with total length length: 235233
## 174 genes
## 170 transcripts
## 191 exons/cds elements
## 61 variations
## 24 other features</code></pre>
<p><code>readGenBank</code> generates a <code>GenBankFull</code> object, which contains the annotations and the origin sequence. <em>genbankr</em> provides methods for the <em>AnnotationDbi</em> API functions for retrieving information from the object.</p>
<p>We can retreive the genes via the function of the same name:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">genes</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span></code></pre></div>
<pre><code>## GRanges object with 174 ranges and 5 metadata columns:
##         seqnames        ranges strand |        type        gene     loctype
##            &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt; &lt;character&gt; &lt;character&gt;
##     [1]   VR1814     1232-2164      + |        gene         RL1      normal
##     [2]   VR1814     2353-4863      - |        gene      RNA2.7      normal
##     [3]   VR1814     5252-5539      - |        gene        RL5A      normal
##     [4]   VR1814     5888-6223      - |        gene         RL6      normal
##     [5]   VR1814     6622-7677      - |        gene      RNA1.2      normal
##     ...      ...           ...    ... .         ...         ...         ...
##   [170]   VR1814 229242-229793      + |        gene        US32      normal
##   [171]   VR1814 229995-230168      + |        gene       US33A      normal
##   [172]   VR1814 230337-230828      + |        gene        US34      normal
##   [173]   VR1814 230822-231016      + |        gene       US34A      normal
##   [174]   VR1814 231986-234349      - |        gene        TRS1      normal
##                           note     gene_id
##                    &lt;character&gt; &lt;character&gt;
##     [1]                   &lt;NA&gt;         RL1
##     [2]                   &lt;NA&gt;      RNA2.7
##     [3]                   &lt;NA&gt;        RL5A
##     [4]                   &lt;NA&gt;         RL6
##     [5]                   &lt;NA&gt;      RNA1.2
##     ...                    ...         ...
##   [170]                   &lt;NA&gt;        US32
##   [171]                   &lt;NA&gt;       US33A
##   [172]                   &lt;NA&gt;        US34
##   [173]                   &lt;NA&gt;       US34A
##   [174] putative beta gene; ..        TRS1
##   -------
##   seqinfo: 1 sequence from GU179289.1 genome</code></pre>
<p>We can also do the same for exons, cds elements, and transcripts (code run but output ommitted for brevity):</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">cds</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>
<span class="fu">exons</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>
<span class="fu">transcripts</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span></code></pre></div>
<p>We can also access elements not in the standard TxDb API, such as variants and features which don’t fit into any of the other categories:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/variants.html">variants</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span>
<span class="fu"><a href="../reference/otherFeatures.html">otherFeatures</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span></code></pre></div>
<p>Furthermore, we can access header-level information via accessors as well:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/gbk-specific-api.html">accession</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "GU179289"</code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/gbk-specific-api.html">vers</a></span><span class="op">(</span><span class="va">gb</span><span class="op">)</span></code></pre></div>
<pre><code>## accession.version         GenInfoID 
##      "GU179289.1"       "270355759"</code></pre>
<p>We can get the seqinfo for a <code>GenBankAnnot</code>/<code>GenBankFull</code> object:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">seqinfo</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span></code></pre></div>
<pre><code>## Seqinfo object with 1 sequence from GU179289.1 genome:
##   seqnames seqlengths isCircular     genome
##   VR1814       235233      FALSE GU179289.1</code></pre>
<p>Finally, we can ge the sequence itself from a <code>GenBankFull</code> object:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">getSeq</span><span class="op">(</span><span class="va">gb</span><span class="op">)</span></code></pre></div>
<pre><code>## DNAStringSet object of length 1:
##      width seq                                              names               
## [1] 235233 CCATTCCGGGCCGCGTGCTGGGT...GCCGCCAGTGCGGGACAGGGCT VR1814</code></pre>
</div>
<div id="low-level-parsing" class="section level1">
<h1 class="hasAnchor">
<a href="#low-level-parsing" class="anchor"></a>Low level parsing</h1>
<p>While use and integration of the Bioconductor machinery is recommended, we also provide low-level parsing capabilities via the workhorse <code>parseGenBank</code> function. This function returns a list structure roughly corresponding to the top-level headings within the genbank format itself:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">pg</span> <span class="op">=</span> <span class="fu"><a href="../reference/parseGenBank.html">parseGenBank</a></span><span class="op">(</span><span class="va">smpfile</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span><span class="op">(</span><span class="va">pg</span>, max.level <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></code></pre></div>
<pre><code>## List of 10
##  $ LOCUS     : chr "LOCUS       GU179289              235233 bp    DNA     linear   VRL 09-MAY-2013"
##  $ FEATURES  :List of 429
##   ..- attr(*, "dim")= int 429
##   ..- attr(*, "dimnames")=List of 1
##  $ ORIGIN    :Formal class 'DNAStringSet' [package "Biostrings"] with 5 slots
##  $ ACCESSION : chr "GU179289"
##  $ COMMENT   : NULL
##  $ DEFINITION: chr "Human herpesvirus 5 strain VR1814, complete genome."
##  $ KEYWORDS  : chr NA
##  $ REFERENCE : NULL
##  $ SOURCE    :List of 3
##  $ VERSION   : Named chr [1:2] "GU179289.1" "270355759"
##   ..- attr(*, "names")= chr [1:2] "accession.version" "GenInfoID"</code></pre>
</div>
<div id="retaining-the-genomic-sequence" class="section level1">
<h1 class="hasAnchor">
<a href="#retaining-the-genomic-sequence" class="anchor"></a>retaining the genomic sequence</h1>
<p>If desired, <code>readGenBank</code> and <code>parseGenBank</code> can omit full sequence of the organism by specifying <code>ret.seq=FALSE</code>. In this case, <code>readGenBank</code> returns a <code>GenBankAnnot</code> object, rather than a <code>GenBankFull</code></p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gbf</span> <span class="op">=</span> <span class="fu"><a href="../reference/readGenBank.html">readGenBank</a></span><span class="op">(</span><span class="va">smpfile</span>, ret.seq <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<pre><code>## Annotations don't have 'locus_tag' label, using 'gene' as gene_id column
## Annotations don't have 'locus_tag' label, using 'gene' as gene_id column</code></pre>
<pre><code>## No exons read from genbank file. Assuming sections of CDS are full exons</code></pre>
<pre><code>## No transcript features (mRNA) found, using spans of CDSs</code></pre>
<pre><code>## Warning in fill_stack_df(feats[!typs %in% c("gene", "exon", "CDS",
## "variation", : Got unexpected multi-value field(s) [ rpt_type ]. The resulting
## column(s) will be of class CharacterList, rather than vector(s). Please contact
## the maintainer if multi-valuedness is expected/meaningful for the listed
## field(s).</code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gbf</span></code></pre></div>
<pre><code>## GenBank Annotations
## Human herpesvirus 5 strain VR1814, complete genome. 
## Accession: GU179289
## 1 Sequence(s) with total length length: 235233
## 174 genes
## 170 transcripts
## 191 exons/cds elements
## 61 variations
## 24 other features</code></pre>
<p>All of the accessor methods discussed above work for <code>GenBankFull</code> objects as they do for <code>GenBankAnnot</code> objects.</p>
</div>
<div id="rtracklayer-style-import" class="section level1">
<h1 class="hasAnchor">
<a href="#rtracklayer-style-import" class="anchor"></a>rtracklayer style import</h1>
<p>We also provide a convenience method for using <em>rtracklayer</em> <code>import</code> style mechanics for reading GenBank files:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gbkfile</span> <span class="op">=</span> <span class="fu"><a href="../reference/GenBankFile-class.html">GenBankFile</a></span><span class="op">(</span><span class="va">smpfile</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning: This class is extending the deprecated RTLFile class from rtracklayer.
## Use BiocFile from BiocIO in place of RTLFile.</code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gb2</span> <span class="op">=</span> <span class="fu">import</span><span class="op">(</span><span class="va">gbkfile</span><span class="op">)</span></code></pre></div>
<pre><code>## Annotations don't have 'locus_tag' label, using 'gene' as gene_id column
## Annotations don't have 'locus_tag' label, using 'gene' as gene_id column</code></pre>
<pre><code>## No exons read from genbank file. Assuming sections of CDS are full exons</code></pre>
<pre><code>## No transcript features (mRNA) found, using spans of CDSs</code></pre>
<pre><code>## Warning in fill_stack_df(feats[!typs %in% c("gene", "exon", "CDS",
## "variation", : Got unexpected multi-value field(s) [ rpt_type ]. The resulting
## column(s) will be of class CharacterList, rather than vector(s). Please contact
## the maintainer if multi-valuedness is expected/meaningful for the listed
## field(s).</code></pre>
</div>
<div id="retrieving-and-parsing-genbank-information-by-versioned-accession" class="section level1">
<h1 class="hasAnchor">
<a href="#retrieving-and-parsing-genbank-information-by-versioned-accession" class="anchor"></a>Retrieving and parsing GenBank information by Versioned Accession</h1>
<p><em>genbankr</em> provides the <code>GBAccession</code> class and constructor for representing versioned Nuccore accession numbers.</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gba</span> <span class="op">=</span> <span class="fu"><a href="../reference/GBAccession-class.html">GBAccession</a></span><span class="op">(</span><span class="st">"U49845.1"</span><span class="op">)</span>
<span class="va">gba</span></code></pre></div>
<pre><code>## GenBank Accession Number(s):  U49845.1</code></pre>
<p>These accession objects can be passed directly to readGenBank:</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/readGenBank.html">readGenBank</a></span><span class="op">(</span><span class="va">gba</span>, partial<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<pre><code>## Loading required namespace: rentrez</code></pre>
<pre><code>## Annotations don't have 'locus_tag' label, using 'gene' as gene_id column
## Annotations don't have 'locus_tag' label, using 'gene' as gene_id column</code></pre>
<pre><code>## genes not available for all CDS ranges, using internal grouping ids</code></pre>
<pre><code>## No exons read from genbank file. Assuming sections of CDS are full exons</code></pre>
<pre><code>## Annotations don't have 'locus_tag' label, using 'gene' as gene_id column
## Annotations don't have 'locus_tag' label, using 'gene' as gene_id column</code></pre>
<pre><code>## GenBank Annotations
## Saccharomyces cerevisiae TCP1-beta gene, partial cds; and Axl2p (AXL2) and Rev7p (REV7) genes, complete cds. 
## Accession: U49845
## 1 Sequence(s) with total length length: 5028
## 2 genes
## 3 transcripts
## 3 exons/cds elements
## 0 variations
## 0 other features</code></pre>
</div>
<div id="retrieving-only-origin-sequence" class="section level1">
<h1 class="hasAnchor">
<a href="#retrieving-only-origin-sequence" class="anchor"></a>Retrieving only origin sequence</h1>
<p><code>genbankr</code> also provides a fastpath for extracting only the sequence of an organism. We can call <code>getSeq</code> on a <code>GenBankFile</code> or <code>GBAccession</code> object</p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">getSeq</span><span class="op">(</span><span class="va">gbkfile</span><span class="op">)</span></code></pre></div>
<pre><code>## DNAStringSet object of length 1:
##      width seq                                              names               
## [1] 235233 CCATTCCGGGCCGCGTGCTGGGT...GCCGCCAGTGCGGGACAGGGCT VR1814</code></pre>
<p>Additionally, we can specify <code>ret.anno = FALSE</code> in <code>parseGenBank</code></p>
<div class="sourceCode" id="cb47"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/parseGenBank.html">parseGenBank</a></span><span class="op">(</span><span class="va">smpfile</span>, ret.anno<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<pre><code>## DNAStringSet object of length 1:
##      width seq                                              names               
## [1] 235233 CCATTCCGGGCCGCGTGCTGGGT...GCCGCCAGTGCGGGACAGGGCT VR1814</code></pre>
</div>
<div id="creating-txdb-objects-from-genbank-annotations" class="section level1">
<h1 class="hasAnchor">
<a href="#creating-txdb-objects-from-genbank-annotations" class="anchor"></a>Creating TxDb objects from genbank annotations</h1>
<p><code>genbankr</code> provides the <code>makeTxDbFromGenBank</code> function, which accepts a <code>GenBankRecord</code> or <code>GBAccession</code> object and returns a <code>TxDb</code> of the annotations.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gbr</span> <span class="op">=</span> <span class="fu"><a href="../reference/readGenBank.html">readGenBank</a></span><span class="op">(</span><span class="va">smpfile</span><span class="op">)</span></code></pre></div>
<pre><code>## Annotations don't have 'locus_tag' label, using 'gene' as gene_id column
## Annotations don't have 'locus_tag' label, using 'gene' as gene_id column</code></pre>
<pre><code>## No exons read from genbank file. Assuming sections of CDS are full exons</code></pre>
<pre><code>## No transcript features (mRNA) found, using spans of CDSs</code></pre>
<pre><code>## Warning in fill_stack_df(feats[!typs %in% c("gene", "exon", "CDS",
## "variation", : Got unexpected multi-value field(s) [ rpt_type ]. The resulting
## column(s) will be of class CharacterList, rather than vector(s). Please contact
## the maintainer if multi-valuedness is expected/meaningful for the listed
## field(s).</code></pre>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tx</span> <span class="op">=</span> <span class="fu"><a href="../reference/makeTxDbFromGenBank.html">makeTxDbFromGenBank</a></span><span class="op">(</span><span class="va">gbr</span><span class="op">)</span>
<span class="va">tx</span></code></pre></div>
<pre><code>## TxDb object:
## # Db type: TxDb
## # Supporting package: GenomicFeatures
## # accession.version: GU179289.1
## # GenInfoID: 270355759
## # genome: GU179289.1
## # source: NCBI (GenBank)
## # definition: Human herpesvirus 5 strain VR1814, complete genome.
## # Db content generated by: genbankr package from Bioconductor
## # genbankr version at creation time: 1.21.0
## # Nb of transcripts: 170
## # Db created by: GenomicFeatures package from Bioconductor
## # Creation time: 2021-07-10 10:25:53 +0000 (Sat, 10 Jul 2021)
## # GenomicFeatures version at creation time: 1.44.0
## # RSQLite version at creation time: 2.2.7
## # DBSCHEMAVERSION: 1.2</code></pre>
</div>
<div id="details-and-caveats" class="section level1">
<h1 class="hasAnchor">
<a href="#details-and-caveats" class="anchor"></a>Details and caveats</h1>
<p>Often times, GenBank files don’t contain exhaustive annotations. For example, files including CDS annotations often do not have separate transcript features. Furthermore, chromosomes are not always named, particularly in organisms that have only one. The details of how genbankr handles such cases are as follows:</p>
<p>In files where CDSs are annotated but individual exons are not, ‘approximate exons’ are defined as the individual contiguous elements within each CDS. Currently, no mixing of approximate and explicitly annotated exons is performed, even in cases where, e.g., exons are not annotated for some genes with CDS annotations.</p>
<p>In files where transcripts are not present, ‘approximate transcripts’ defined by the ranges spanned by groups of exons are used. Currently, we do not support generating approximate transcripts from CDSs in files that contain actual transcript annotations, even if those annotations do not cover all genes with CDS/exon annotations.</p>
<p>Features (gene, cds, variant, etc) are assumed to be contained within the most recent previous source feature (chromosome/physical piece of DNA). Chromosome name for source features (seqnames in the resulting GRanges}/VRanges is determined as follows:</p>
<ol style="list-style-type: decimal">
<li>The ‘chromosome’ attribute, as is (e.g., “chr1”);</li>
<li>the ‘strain’ attribute, combined with auto-generated count (e.g., “VR1814:1”);</li>
<li>the ‘organism’ attribute, combined with auto-generated count (e.g.“Human herpesvirus 5:1”)</li>
</ol>
<p>Some GenBank files do not include origin sequence. In these cases, variation features are not supported, as there is no self-contained way to determine reference sequence and the features themselves typically contain only alt information (if that). In the case of files containing variation features but no origin sequence, those features are ignored with a warning.</p>
<p>Currently some information about from the header of a GenBank file, primarily reference and author based information, is not captured and returned. Please contact the maintainer if you have a direct use-case for this type of information.</p>
</div>
<div id="performance" class="section level1">
<h1 class="hasAnchor">
<a href="#performance" class="anchor"></a>Performance</h1>
<p>We have taken pains to make the genbankr parser as effcient as easily possible. On our local machines, a 19MB genbank file takes 2-3 minutes to be parsed. That said, this package is not tested and likely is not suitable for processing extremely large genbank files. We suggest obtaining the annotations in a different format in such cases.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><!-- begin footer --><div class="footer">
    <div class="container">
        <div class="row start bottom-8">
            <div class="col-xs-2"> <img id="footerlogo" src="https://ropensci.org/img/icon_short_white.svg">
</div>
            <div class="col-xs-10">
                <div class="row">
                    <div class="col-md-4 col-xs-6">
                        <a href="http://github.com/ropensci" target="_blank"><div class="icon fab fa-github"></div></a>
                        <a href="http://github.com/ropenscilabs" target="_blank"><div class="icon fa fa-flask"></div></a>
                        <a href="http://twitter.com/ropensci" target="_blank"><div class="icon fab fa-twitter"></div></a>
                        <a href="http://vimeo.com/ropensci" target="_blank"><div class="icon fab fa-vimeo"></div></a>
                    </div>
                </div>
                <div class="row top-4">
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">About</h5>
                            <li><a href="https://ropensci.org/about">About rOpenSci</a></li>
                            <li><a href="https://ropensci.org/software-review">Software Review Process</a></li>
                            <li><a href="https://ropensci.org/about#team">Our Team</a></li>
                            <li><a href="https://ropensci.org/careers">Jobs</a></li>
                            <li><a href="https://ropensci.org/donate">Donate</a></li>
                            <li><a href="https://ropensci.org/contact">Contact Us</a></li>
                        </ul>
</div>
                    <div class="col-md-3 col-sm-4">
                        <ul>
<h5 class="bottom-2">Community</h5>
                            <li><a href="https://ropensci.org/community/">Our Community</a></li>
                            <li><a href="https://ropensci.org/commcalls/">Community calls</a></li>
                            <li><a href="https://ropensci.org/events/">Events</a></li>
                            <li><a href="https://discuss.ropensci.org/">Join the Discussion</a></li>
                            <li><a href="https://ropensci.org/code-of-conduct">Code of conduct</a></li>
                        </ul>
</div>
                    <div class="col-md-2 col-sm-4">
                        <ul>
<h5 class="bottom-2">Resources</h5>
                            <li><a href="https://ropensci.org/packages/">Packages</a></li>
                            <li><a href="https://ropensci.org/usecases/">Use Cases</a></li>
                            <li><a href="https://ropensci.org/talks-papers/">Talks &amp; Publications</a></li>
                            <li><a href="https://docs.ropensci.org/">Documentation</a></li>
                            <li><a href="https://ropensci.org/how-to-cite-ropensci/">Cite rOpenSci</a></li>
                        </ul>
</div>
                    <div class="col-md-4 col-xs-12">
                        <h5 class="bottom-2"></h5>

                        <p>rOpenSci is a fiscally sponsored project of <a href="http://numfocus.org" target="_blank">NumFOCUS</a></p>
                        <a href="http://numfocus.org" target="_blank"></a>

                    </div>
                </div>
            </div>
        </div>
    </div>
</div>
<!-- / end footer -->

      </footer>
</div>

  
<script src="https://cdnjs.cloudflare.com/ajax/libs/docsearch.js/2.6.1/docsearch.min.js" integrity="sha256-GKvGqXDznoRYHCwKXGnuchvKSwmx9SRMrZOTh2g4Sb0=" crossorigin="anonymous"></script><script>
  docsearch({
    
    
    apiKey: '799829e946e1f0f9cd5b5a782c6316b9',
    indexName: 'ropensci-genbankr',
    inputSelector: 'input#search-input.form-control',
    transformData: function(hits) {
      return hits.map(function (hit) {
        hit.url = updateHitURL(hit);
        return hit;
      });
    }
  });
</script>
</body>
</html>
